types/nullable_test.go

```diff
@@ -0,0 +1,248 @@
+}
+
+func TestMixed(t *testing.T) {
+	type testCase struct {
+		name      string
+		jsonInput []byte
+		assert    func(obj TestComplex, t *testing.T)
+	}
+	tests := []testCase{
+		{
+			name:      "empty json input",
+			jsonInput: []byte(`{}`),
+			assert: func(obj TestComplex, t *testing.T) {
+				assert.Equalf(t, false, obj.SimpleInt.Value.ReplicaCount.IsSet(), "replica count should not be set")
+				assert.Equalf(t, false, obj.SimpleInt.Value.ReplicaCount.IsNull(), "replica count should not be null")
+				assert.Equalf(t, false, obj.SimpleString.Value.Name.IsSet(), "name should not be set")
+				assert.Equalf(t, false, obj.SimpleString.Value.Name.IsNull(), "name should not be null")
+				assert.Equalf(t, false, obj.StringList.IsSet(), "string list should not be set")
+				assert.Equalf(t, false, obj.StringList.IsNull(), "string list should not be null")
+			},
+		},
+
+		{
+			name:      "replica count having non null value",
+			jsonInput: []byte(`{"simple_int":{"replicaCount":1}}`),
+			assert: func(obj TestComplex, t *testing.T) {
+				assert.Equalf(t, false, obj.SimpleInt.Value.ReplicaCount.IsNull(), "replica count should NOT be null")
+				assert.Equalf(t, true, obj.SimpleInt.Value.ReplicaCount.IsSet(), "replica count should be set")
+				assert.Equalf(t, false, obj.SimpleString.Value.Name.IsSet(), "name should NOT be set")
+				assert.Equalf(t, false, obj.SimpleString.Value.Name.IsNull(), "name should NOT be null")
+				gotValue, isNull := obj.SimpleInt.Value.ReplicaCount.Get()
+				assert.Equalf(t, false, isNull, "replica count should NOT be null")
+				assert.Equalf(t, 1, gotValue, "replica count should be 1")
+			},
+		},
+
+		{
+			name:      "string list having null value",
+			jsonInput: []byte(`{"string_list": null}`),
+			assert: func(obj TestComplex, t *testing.T) {
+				assert.Equalf(t, true, obj.StringList.IsSet(), "string_list should be set")
+				assert.Equalf(t, true, obj.StringList.IsNull(), "string_list should be null")
+			},
+		},
+
+		{
+			name:      "string list having non null value",
+			jsonInput: []byte(`{"string_list": ["foo", "bar"]}`),
+			assert: func(obj TestComplex, t *testing.T) {
+				assert.Equalf(t, true, obj.StringList.IsSet(), "string_list should be set")
+				assert.Equalf(t, false, obj.StringList.IsNull(), "string_list should not be null")
+				gotStringList, isNull := obj.StringList.Get()
+				assert.Equalf(t, false, isNull, "string_list should not be null")
+				assert.Equalf(t, []string{"foo", "bar"}, gotStringList, "string_list should have the values as provided in the jSON")
+
+			},
+		},
+
+		{
+			name:      "set string list having empty value",
+			jsonInput: []byte(`{"string_list":[]}`),
+			assert: func(obj TestComplex, t *testing.T) {
+				assert.Equalf(t, true, obj.StringList.IsSet(), "string_list should be set")
+				assert.Equalf(t, false, obj.StringList.IsNull(), "string_list should not be null")
+				gotStringList, isNull := obj.StringList.Get()
```

Comment:
Mind adding newlines for readability?
```suggestion

				gotStringList, isNull := obj.StringList.Get()
```